store/postgres/src/store.rs

```diff
@@ -323,126 +321,264 @@ impl Store {
+                ValueType::Int => "::bigint",
+                ValueType::String => "",
+            };
+            diesel_query = diesel_query.order(
+                sql::<Text>("(data ->")
+                    .bind::<Text, _>(order_attribute)
+                    .sql("->> 'data')")
+                    .sql(cast_type)
+                    .sql(" ")
+                    .sql(direction)
+                    .sql(" NULLS LAST"),
+            );
+        }
+
+        // Add range filter to query
+        if let Some(range) = query.range {
+            diesel_query = diesel_query
+                .limit(range.first as i64)
+                .offset(range.skip as i64);
+        }
+
+        // Record debug info in case of error
+        let diesel_query_debug_info = debug_query(&diesel_query).to_string();
+
+        // Process results; deserialize JSON data
+        diesel_query
+            .load::<serde_json::Value>(conn)
+            .map(|values| {
+                values
+                    .into_iter()
+                    .map(|value| {
+                        let parse_error_msg = format!("Error parsing entity JSON: {:?}", value);
+                        serde_json::from_value::<Entity>(value).expect(&parse_error_msg)
+                    })
+                    .collect()
+            })
+            .map_err(|e| {
+                QueryExecutionError::ResolveEntitiesError(format!(
+                    "{}, query = {:?}",
+                    e, diesel_query_debug_info
+                ))
+            })
+    }
+
+    /// Applies a set operation in Postgres.
+    fn apply_set_operation(
+        &self,
+        conn: &PgConnection,
+        key: EntityKey,
+        data: Entity,
+        event_source: EventSource,
+    ) -> Result<(), StoreError> {
+        use db_schema::entities;
 
         // Load the entity if exists
-        let existing_entity =
-            self.get_entity(conn, op_subgraph_id, op_entity_type, op_entity_id)?;
+        let existing_entity = self
+            .get_entity(conn, &key.subgraph_id, &key.entity_type, &key.entity_id)
+            .map_err(Error::from)?;
 
         // Apply the operation
+        let operation = EntityOperation::Set {
+            key: key.clone(),
+            data,
+        };
         let updated_entity = operation.apply(existing_entity);
         let updated_json: serde_json::Value =
             serde_json::to_value(&updated_entity).map_err(|e| {
                 format_err!(
                     "Failed to set entity ({}, {}, {}) as setting it would break it: {}",
-                    op_subgraph_id,
-                    op_entity_type,
-                    op_entity_id,
+                    key.subgraph_id,
+                    key.entity_type,
+                    key.entity_id,
                     e
                 )
             })?;
 
         // Either add or update the entity in Postgres
-        insert_into(entities)
+        insert_into(entities::table)
```

Comment:
Isn't the `entities` DSL available any more?

Reply:
Yeah, I just got frustrated with having column names conflict with local variable names -- they're in the same namespace when importing the dsl. I can revert if you like

Reply:
Nah, it's fine.